src/target/stm32mp15.c

```diff
@@ -96,6 +105,51 @@ static bool stm32mp15_ident(target_s *const target, const bool cortexm)
 	return true;
 }
 
+static bool stm32mp15_cmd_swo(target_s *target, int argc, const char **argv)
+{
+	(void)argc;
+	(void)argv;
+	/* TODO: argv parsing for mode and baudrate */
+	adiv5_access_port_s *const ap_apbd = (adiv5_access_port_s *)target->target_storage;
+	/* Pin Protocol: change Manchester to UART */
+	uint32_t sppr = 0;
+	adiv5_mem_read(ap_apbd, &sppr, SWO_SPPR, 4);
+	sppr &= ~(0x3U);
+	sppr |= 0x2U;
+	adiv5_mem_write(ap_apbd, SWO_SPPR, &sppr, 4);
+
+	/* 
+	 * Prescaler: set to fixed 66; trace clk freq of 133/(66+1) gives ~2Mbaud (+-0.7%)
+	 * assuming AXI clk of 266 and default divisor of 2
+	 * Or, if you are not restricted by swlink 2.25M, set to fixed 32; 133/(32+1) is ~4Mbaud (+-0.7%)
+	 */
+	uint32_t acpr = 0;
+	adiv5_mem_read(ap_apbd, &acpr, SWO_ACPR, 4);
+	acpr = 32;
+	adiv5_mem_write(ap_apbd, SWO_ACPR, &acpr, 4);
+
+	return true;
+}
+
+static void stm32mp15_setup_apbd_ap(target_s *const target)
```

Comment:
Given how the setup functions are named (and we agree with this naming, though aren't sure where the `d` comes from in `apbd` here?), we suggest being absolutely consistent below - `apb_ap`, `axi_ap`, so the code's a little easier to follow. Either that or using the AP number instead of bus type for the naming - `ap0`, `ap1`? That one's up to you which you think would be more beneficial here.